packages/cosmwasm/src/signingcosmwasmclient.ts

```diff
@@ -663,6 +663,75 @@ export class SigningCosmWasmClient extends CosmWasmClient {
     return this.broadcastTxSync(txBytes);
   }
 
+  private async calculateFeeForTransaction(
+    signerAddress: string,
+    messages: readonly EncodeObject[],
+    memo: string,
+    fee: "auto" | number,
+  ): Promise<StdFee> {
+    const gasEstimation = await this.simulate(signerAddress, messages, memo);
+    const multiplier = typeof fee === "number" ? fee : this.defaultGasMultiplier;
```

Comment:
At this point we know the gas limit, right? Everything else below is just about the price.

Adding

```ts
const gasLimit = Math.ceil(gasEstimation * multiplier)
```

here would remove the 4x `Math.round(gasEstimation * multiplier)` from below and increase readability.